### PR TITLE
DB-7035 Fixes on ACL layer 

### PR DESCRIPTION
**Pull request by dgomezferro**

### CODE REVIEW

**Comment by OlegMazurov** [comment]
com.splicemachine.pipeline.mem.DirectBulkWriter.java.write() needs to be fixed to compile on mem

**Comment by jyuanca** [comment]
Why we need a partitionAdapterCache and can't use partitionCache?

**Comment by dgomezferro** [comment]
@jyuanca if we use partitionCache it would get populated by AdapterPartitions and ClientPartitions, the latter don't know how to proxy the methods through the JDBC connection and they would fail when reused at a later point. It might be possible to make sure we only generate AdapterPartitions, but I found it easier to separate the two.

File: splice_access_api/src/main/java/com/splicemachine/access/configuration/SConfigurationImpl.java [hunk 1]
```diff
@@ -276,7 +277,10 @@ public boolean getAuthenticationTokenDebugConnections() {
     public int getAuthenticationTokenMaxConnections() {
         return authenticationTokenMaxConnections;
     }
-
+    @Override
+    public int getAuthenticationTokenPermissionCacheSize() {
```
Comment (yxia92) [inline splice_access_api/src/main/java/com/splicemachine/access/configuration/SConfigurationImpl.java]: This configuration parameter does not seem to be used anywhere. Did I miss anything?

**Comment by dgomezferro** [comment]
@yxia92 I forgot to create the -ee PR, thanks for noticing! Please review it here: https://github.com/splicemachine/spliceengine-ee/pull/218

File: splice_access_api/src/main/java/com/splicemachine/access/configuration/SConfigurationImpl.java (hunk 1 above)
Comment (dgomezferro) [reply splice_access_api/src/main/java/com/splicemachine/access/configuration/SConfigurationImpl.java]: @yxia92 It's used on the -ee side, I forgot to create the PR, sorry!

File: hbase_storage/hbase1.1.8/src/main/java/com/splicemachine/storage/Hbase10PartitionCache.java [hunk 2]
```diff
@@ -63,5 +64,27 @@ public void configure(SConfiguration configuration){
     public void put(TableName tableName, List<Partition> partitions) throws IOException {
         partitionCache.put(tableName,partitions);
     }
+
+    @Override
+    public void invalidateAdapter(TableName tableName) throws IOException {
+        partitionAdapterCache.invalidate(tableName);
+        ((ClusterConnection)HBaseConnectionFactory.getInstance(config).getConnection()).clearRegionCache(tableName);
```
Comment (yxia92) [inline hbase_storage/hbase1.1.8/src/main/java/com/splicemachine/storage/Hbase10PartitionCache.java]: Need to change the use of HConnection in the function invalidate() to ClusterConnection to be consistent with invalidateAdapter. We got into compilation error due to this for the special branch-2.5-1820c.

Comment (dgomezferro) [reply hbase_storage/hbase1.1.8/src/main/java/com/splicemachine/storage/Hbase10PartitionCache.java]: @yxia92 I'm away from my computer, could you (or somebody else) fix it, if merging this is urgent? Thanks

Comment (yxia92) [reply hbase_storage/hbase1.1.8/src/main/java/com/splicemachine/storage/Hbase10PartitionCache.java]: I can push a fix for the compilation issue.

**Pull request merged**